### DISCUSSION TITLE
Provide setting to disable updates

### DISCUSSION THREAD

**Discussion by stevehipwell** [bottlerocket-os/bottlerocket#1804] [Q&A] [answered]
**What I'd like:**
I'd like to be able to disable updates (or equivalent) so that all nodes started from a launch template version are identical and can't be modified. This means that either the AMI ID or a combination of the AMI ID and a configuration value need to define the state. We use immutable infrastructure so will replace the node rather than updating it in place so it doesn't make sense to use bandwidth pulling updates.

**Any alternatives you've considered:**
I guess not running the update command would have the same effect, but I'd like confirmation?

**Comment by samuelkarp** [comment, marked as answer]
Hello!

> I guess not running the update command would have the same effect, but I'd like confirmation?

By default, nothing installed in the operating system itself will initiate an update.  The [Bottlerocket update operator for Kubernetes](https://github.com/bottlerocket-os/bottlerocket-update-operator) and the [Bottlerocket ECS updater](https://github.com/bottlerocket-os/bottlerocket-ecs-updater/) can be optionally used to automatically initiate updates for nodes in a cluster.

> I'd like to be able to disable updates (or equivalent) so that all nodes started from a launch template version are identical and can't be modified. This means that either the AMI ID or a combination of the AMI ID and a configuration value need to define the state.

In addition to refraining from initiating updates, you may also use the [`settings.updates.version-lock` setting](https://github.com/bottlerocket-os/bottlerocket#updates-settings) to lock to a specific version.

Let us know if you have any additional questions.

**Reply by stevehipwell** [reply]
@samuelkarp it looks like I dropped the ball twice on this one; firstly I didn't spot the discussions tab, and then if I'd only waited an hour or so I'd have seen the [bottlerocket-a-year-in-the-life](https://aws.amazon.com/blogs/containers/bottlerocket-a-year-in-the-life/) blog post which would have answered my question.

I do have a follow up question regarding the versioning and releases; are the releases we see in GitHub the only versions or are there other versions available? I suspect (and hope) it's the former but would like clarification.

**Reply by bcressey** [reply]
> I do have a follow up question regarding the versioning and releases; are the releases we see in GitHub the only versions or are there other versions available? I suspect (and hope) it's the former but would like clarification.

The releases in GitHub are the only first party versions available. We build directly from the unmodified Git commit that shows up in the AMI names and in `/etc/os-release` as the `BUILD_ID` field.
